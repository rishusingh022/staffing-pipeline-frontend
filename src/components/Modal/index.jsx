import React from 'react';
import PropTypes from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faXmark } from '@fortawesome/free-solid-svg-icons';

const Modal = ({ children, open }) => {
  const [isOpen, setIsOpen] = React.useState(open);

  const handleClose = () => {
    setIsOpen(false);
  };

  return (
    isOpen && (
      <div className="z-10 fixed inset-0 flex justify-center items-center">
        <div className="absolute inset-0 bg-black/[.85]"></div>
        <div className="relative z-1 bg-white p-10 w-[40rem] h-[9rem]">
          <div
            className="absolute top-3 right-3 bg-transparent border-none cursor-pointer w-4 h-4 flex justify-center items-center"
            onClick={handleClose}>
            <FontAwesomeIcon icon={faXmark} />
          </div>
          {children}
        </div>
      </div>
    )
  );
};
Modal.propTypes = {
  children: PropTypes.node.isRequired,
  open: PropTypes.bool.isRequired,
};

export default Modal;
