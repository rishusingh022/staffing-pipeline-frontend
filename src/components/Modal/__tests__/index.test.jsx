import React from 'react';
import { render } from '@testing-library/react';
import Modal from '..';

const onClose = jest.fn();

describe('Modal', () => {
  it('should render correctly if isOpen prop is true', () => {
    const { asFragment } = render(<Modal {...{ isOpen: true, onClose }} />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('should not render if isOpen prop is false', () => {
    const { asFragment } = render(<Modal {...{ isOpen: false, onClose }} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
