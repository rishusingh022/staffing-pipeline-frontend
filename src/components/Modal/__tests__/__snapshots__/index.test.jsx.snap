// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should not render if isOpen prop is false 1`] = `
<DocumentFragment>
  <div
    class="z-10 fixed inset-0 flex justify-center items-center"
  >
    <div
      class="absolute inset-0 bg-black/[.85]"
    />
    <div
      class="relative z-1 bg-white p-10 w-[40rem] h-[9rem]"
    >
      <div
        class="absolute top-3 right-3 bg-transparent border-none cursor-pointer w-4 h-4 flex justify-center items-center"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-xmark "
          data-icon="xmark"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 320 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M310.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L160 210.7 54.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L114.7 256 9.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L160 301.3 265.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L205.3 256 310.6 150.6z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Modal should render correctly if isOpen prop is true 1`] = `
<DocumentFragment>
  <div
    class="z-10 fixed inset-0 flex justify-center items-center"
  >
    <div
      class="absolute inset-0 bg-black/[.85]"
    />
    <div
      class="relative z-1 bg-white p-10 w-[40rem] h-[9rem]"
    >
      <div
        class="absolute top-3 right-3 bg-transparent border-none cursor-pointer w-4 h-4 flex justify-center items-center"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-xmark "
          data-icon="xmark"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 320 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M310.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L160 210.7 54.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L114.7 256 9.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L160 301.3 265.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L205.3 256 310.6 150.6z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
